go/vt/orchestrator/inst/analysis_dao.go

```diff
@@ -308,7 +308,7 @@ func GetReplicationAnalysis(clusterName string, hints *ReplicationAnalysisHints)
 		) AS count_distinct_logging_major_versions
 	FROM
 		vitess_tablet
-		LEFT JOIN database_instance primary_instance ON (
+		JOIN database_instance primary_instance ON (
```

Comment:
Why did this need to change?

Reply:
The `database_instance` table contains critical information to be able to run any failover scenario. I encountered a bug wherein a vttablet instance was discovered and the `vitess_tablet` table was populated but before it's cluster information could be added to the `database_instance` we tried to run a failover. This tablet had no record in the `database_instance` table. This led to the LEFT JOIN filling all the values from that table with nils even the cluster name. So we ended up concluding that this tablet was the only tablet in its cluster and run a `clusterHasNoPrimary` failover.
I realised that we should only be running failover on tablets for which we have information in both the tables hence changed the LEFT JOIN to a normal JOIN and this fixed the bug. While discovery we populate both the tables but there is short time where we have populated the `vitess_tablet` table after reading the topo but have not populated the `database_instance` table since we are reading variables from the underlying mysql instance like the replication_lag, etc. We should discount these instances from running failovers until their information has been properly stored.